controllers/storageautoscaler/storageautoscaler_controller_test.go

```diff
@@ -0,0 +1,70 @@
+package storageautoscaler
+
+import (
+	"testing"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func TestCheckIfScalingNotRequired(t *testing.T) {
+	usage := 0.055
+	threshold := 6
+
+	required := checkIfScalingRequired(usage, threshold)
+	assert.False(t, required)
+
+	usage = 0.05
+	threshold = 5
+
+	required = checkIfScalingRequired(usage, threshold)
+	assert.True(t, required)
+
+	usage = 0.05
+	threshold = 4
+
+	required = checkIfScalingRequired(usage, threshold)
+	assert.True(t, required)
+}
+
+func TestCalculateExpectedOsdSizeAndCount(t *testing.T) {
```

Comment:
Please add other cases, currently I only see one case for horizontal scaling. Cases are needed for vertical scaling, as well as negative cases that confirm that resizing is not triggered when we are not above the threshold.

Reply:
Added a check for `threshold` function separately